main.go

```diff
@@ -42,14 +41,15 @@ func main() {
 	log.Info(fmt.Sprint("Peerster IP Address: ", *peersterAddr))
 	log.Info(fmt.Sprint("Peerster Name: ", *name))
 	log.Info(fmt.Sprint("Number of bits in one node's id: ", *m))
-	log.Info(fmt.Sprint("Number of nodes in the successor list: ", *r))
 
 	listener, err := net.Listen("tcp", *peersterAddr)
 	if err != nil {
 		log.Fatal(fmt.Sprintf("listening to %s failed: %v", *peersterAddr, err))
 	}
 
-	chordNode, err := chord.NewChordNode(listener)
+	chordNode, err := chord.NewChordNode(listener, chord.ChordConfig{
+		NumOfBitsInID: int32(*m),
```

Comment:
You can remove cast here (use simple `int`), so that you can avoid cast here `next >= int(m)` (FixFingersDaemon.go).  
   
`m` can be `4,8,12,..,256`